rfd/0204-aws-iam-roles-anywhere.md

```diff
@@ -0,0 +1,769 @@
+- Subject fields as `PrincipalTag: aws:PrincipalTag/x509Subject/<field>` (eg CN maps to `PrincipalTag: aws:PrincipalTag/x509Subject/CN`)
+- Issuer fields as `PrincipalTag: aws:PrincipalTag/x509Issuer/<field>`
+- Subject Alternative Name (SAN) as `PrincipalTag: aws:PrincipalTag/x509SAN/<field>`
+
+Of the above, only some of them will be set by Teleport:
+- Subject Common Name: Teleport user that requested the access
+- Issuer Common Name: Teleport cluster name
+- Not After: this will be set based on the user's current session expiration
+
+### AWS Session creation
+After generating the certificate, Teleport will call `rolesanywhere.CreateSession` and exchange the certificate for AWS credentials.
+
+This call will not be explicit, but handled by the [rolesanywhere-credential-helper](https://github.com/aws/rolesanywhere-credential-helper) tool from AWS.
+
+The `rolesanywhere.CreateSession` call accepts a `durationSeconds` which indicates for how long the AWS Session will be valid.
+This durantion, per AWS documentation, cannot be higher than 12 hours, nor less than 15 minutes.
+
+When the Teleport User session is between 15 minutes and 12 hours, the AWS Session will have the same expiration.
+
+When the Teleport User session is lower than 15 minutes, the user will receive an error and is asked to login again.
+This is already the behavior on existing AWS Access flows. See [#46551](https://github.com/gravitational/teleport/issues/46551).
+
+When the Teleport User session is higher than 12 hours, the AWS Session will be at most 12 hours.
+
+### AWS Roles requirements for usage with IAM Roles Anywhere
+AWS IAM Roles must be accessible from Roles Anywhere service principal, which requires a [custom Trust Policy](https://docs.aws.amazon.com/rolesanywhere/latest/userguide/trust-model.html#trust-policy).
+
+For IAM Roles to be accessible from IAM Roles Anywhere, they must have the following trust policy:
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Principal": {
+          "Service": "rolesanywhere.amazonaws.com"
+        },
+        "Action": [
+          "sts:AssumeRole",
+          "sts:TagSession",
+          "sts:SetSourceIdentity"
+        ],
+        "Condition": {
+          "ArnEquals": {
+            "aws:SourceArn": [
+              "arn:aws:rolesanywhere:<region>:<account>:trust-anchor/<trust-anchor-id>"
+            ]
+          }
+        }
+      }
+    ]
+  }
+```
+
+Note: The `Condition` section is optional, and will ensure only the referenced Trust Anchor can use this IAM Role.
+
+### Integration with end user tools
+Users will login using `tsh apps login --aws-role <Role ARN> <App Name>`.
+This command does two things:
+- fetches and stores the credentials into the application certificate stored under `~/.tsh/`
+- modifies the `~/.aws/config` file to create/update the AWS configuration profile
```

Comment:
will `~/.aws/config` be cleaned up during logout? (similar to how we handle the kube config)

Reply:
Nice catch, I'll definitely add that to the RFD, thank you.